packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1535 @@
+        builder.hasMultipleLines) {
+      final String prefix = prefixChildren.trimRight();
+      if (prefix.isNotEmpty)
+        builder.writeRaw('$prefix${config.lineBreak}');
+    }
+
+    if (children.isNotEmpty && config.showChildren) {
+      if (config.isBlankLineBetweenPropertiesAndChildren &&
+          properties.isNotEmpty &&
+          children.first.textTreeConfiguration.isBlankLineBetweenPropertiesAndChildren) {
+        builder.write(config.lineBreak);
+      }
+
+      for (int i = 0; i < children.length; i++) {
+        final DiagnosticsNode child = children[i];
+
+        final TextTreeConfiguration childConfig = _childTextConfiguration(child, config);
+        if (i == children.length - 1) {
+          final String lastChildPrefixLineOne =
+              '$prefixChildren${childConfig.prefixLastChildLineOne}';
+          if (child == null) {
+            builder.writeRawLine('$lastChildPrefixLineOne<null>');
+            continue;
+          }
+          builder.writeRawLine(child.toStringDeep(
+              lastChildPrefixLineOne,
+              '$prefixChildren${childConfig.childLinkSpace}${childConfig.prefixOtherLines}'));
+          if (childConfig.footer.isNotEmpty)
+            builder.writeRaw('$prefixChildren${childConfig.childLinkSpace}${childConfig.footer}');
+
+        } else {
+          final TextTreeConfiguration nextChildStyle =
+              _childTextConfiguration(children[i + 1], config);
+
+          final String childPrefixLineOne = '$prefixChildren${childConfig.prefixLineOne}';
+          final String childPrefixOtherLines ='$prefixChildren${nextChildStyle.linkCharacter}${childConfig.prefixOtherLines}';
+
+          if (child == null) {
+            builder.writeRawLine('$childPrefixLineOne<null>');
+            continue;
+          }
+          builder.writeRawLine(
+              child.toStringDeep(childPrefixLineOne, childPrefixOtherLines));
+          if (childConfig.footer.isNotEmpty)
+            builder.writeRaw('$prefixChildren${nextChildStyle.linkCharacter}${childConfig.footer}');
+        }
+      }
+    }
+    return builder.toString();
+  }
+}
+
+/// Property which supports specifying whether the string is value is enclosed
+/// in quotes when displayed.
+class StringProperty extends DiagnosticsProperty<String> {
+  StringProperty(String name, String value, {
+    String description,
+    bool showName: true,
+    Object defaultValue: kNoDefaultValue,
+    bool hidden: false,
+    this.quoted: false,
+    String emptyDescription,
+  }) : super(
+    name,
+    value,
+    description: description,
+    defaultValue: defaultValue,
+    showName: showName,
+    hidden: hidden,
+    emptyDescription: emptyDescription,
+  );
+
+  /// Whether the description is enclosed in double quotes.
+  final bool quoted;
+
+  @override
+  String valueToString() {
+    final String text = _description ?? value;
+    return (quoted && text != null) ? '"$text"' : text.toString();
```

Comment:
Actually using JSON.encode has negative consequences. Now you will see
```              ' ╘═╦══ text ═══\n'
              '   ║ TextSpan:\n'
              '   ║   "I polished up that handle so carefullee\\nThat now I am the Ruler\n'
              '   ║     of the Queen\'s Navee!"\n'
              '   ╚═══════════\n'
```
which is technically correct but less readable. Would probably need to hybrid escape where \n is left alone but " is still escaped correctly. Added a TODO and switched back to quoting for clarity but not correct escaping.